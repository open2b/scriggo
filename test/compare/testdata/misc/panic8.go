// paniccheck

// Test a runtime error propagation.

package main

import (
	"runtime"

	"github.com/open2b/scriggo/test/compare/testpkg"
)

func main() {
	defer func() {
		r := recover()
		if _, ok := r.(runtime.Error); !ok {
			panic("BUG")
		}
		panic(r)
	}()
	testpkg.CallFunction(f)
}

func f() {
	var a = 0
	_ = 1 / a
}
