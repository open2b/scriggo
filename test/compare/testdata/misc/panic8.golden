panic: runtime error: integer divide by zero